### PR TITLE
Swift: Add CSV extension points to the encryption queries.

### PR DESCRIPTION
**Pull request by geoffw0**
Add MaD/CSV extension points to the eight encryption queries.  Use them in two of the queries (I intend to take this a bit further at some point).  Fix a few other bits and pieces around sinks for better reliability and consistency.

### CODE REVIEW

File: swift/ql/lib/codeql/swift/security/ConstantSaltExtensions.qll [hunk 1]
```diff
@@ -49,8 +49,12 @@ private class CryptoSwiftSaltSink extends ConstantSaltSink {
  */
 private class RnCryptorSaltSink extends ConstantSaltSink {
   RnCryptorSaltSink() {
-    exists(ClassOrStructDecl c, MethodDecl f, CallExpr call |
-      c.getName() = ["RNCryptor", "RNEncryptor", "RNDecryptor"] and
+    exists(NominalTypeDecl c, MethodDecl f, CallExpr call |
+      c.getName() =
```
Comment (MathiasVP) [inline swift/ql/lib/codeql/swift/security/ConstantSaltExtensions.qll:53]: Should this be `getFullName` like the other cases?

File: swift/ql/lib/codeql/swift/security/ConstantPasswordExtensions.qll [hunk 2]
```diff
@@ -50,8 +50,12 @@ private class DefaultConstantPasswordSink extends ConstantPasswordSink {
 private class RnCryptorPasswordSink extends ConstantPasswordSink {
   RnCryptorPasswordSink() {
     // RNCryptor (labelled arguments)
-    exists(ClassOrStructDecl c, MethodDecl f, CallExpr call |
-      c.getName() = ["RNCryptor", "RNEncryptor", "RNDecryptor"] and
+    exists(NominalTypeDecl c, MethodDecl f, CallExpr call |
+      c.getName() =
```
Comment (MathiasVP) [inline swift/ql/lib/codeql/swift/security/ConstantPasswordExtensions.qll:54]: Should this be `getFullName` like the other cases?

**Review by MathiasVP — APPROVED** [review]
A couple of questions, but otherwise this LGTM!

File: swift/ql/lib/codeql/swift/security/ConstantSaltExtensions.qll (hunk 1 above)
Comment (geoffw0) [reply swift/ql/lib/codeql/swift/security/ConstantSaltExtensions.qll]: Oh, well spotted!

Fixed both cases.

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**